src/core/connection.h

```diff
@@ -1390,3 +1390,54 @@ QuicConnParamGet(
     _Out_writes_bytes_opt_(*BufferLength)
         void* Buffer
     );
+
+//
+// Get the max MTU for a specific path.
+//
+_IRQL_requires_max_(DISPATCH_LEVEL)
+inline
+uint16_t
+QuicConnGetMaxMtuForPath(
+    _In_ QUIC_CONNECTION* Connection,
+    _In_ QUIC_PATH* Path
+    )
+{
+    uint16_t LocalMtu = CxPlatSocketGetLocalMtu(Path->Binding->Socket);
```

Comment:
This may be a non-trivial query in the future. Might be a good idea to cache this whole calculation.

Reply:
If we want to do any caching, I'd want to cache it at the socket layer. I don't think caching it at the connection is a good idea.

Reply:
The size is (eventually) per-path, not per-socket, so we can't cache at the socket. It has to be cached at the path.